### PR TITLE
feat: beta upgrade notice

### PR DESCRIPTION
**Pull request by nud3l**

### CODE REVIEW

File: src/assets/locales/en/translation.json [hunk 1]
```diff
@@ -520,5 +520,15 @@
   },
   "testnet": {
     "warning": "Notice: All assets in the beta testnet are for testing purposes only and have no real value."
+  },
+  "maintenance": {
+    "type": {
+      "scheduled": "Scheduled maintenance on ",
+      "unplanned": "Unplanned maintenanceon "
```
Comment (anton-karlovskiy) [inline src/assets/locales/en/translation.json:527]: Do we need trailing space?

File: src/components/MaintenanceBanner/index.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import { useTranslation } from 'react-i18next';
+import clsx from 'clsx';
+import { formatDateTime } from 'common/utils/utils';
+
+interface Message {
+  startTime: Date,
+  endTime: Date,
+  type: string,
+  reason: string,
+  link: string
```
Comment (anton-karlovskiy) [inline src/components/MaintenanceBanner/index.tsx:11]: I guess the syntax should be:
```
startTime: Date;
endTime: Date;
type: string;
reason: string;
link: string;
```
Please correct me if I'm wrong.

File: src/components/MaintenanceBanner/index.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
+
+import { useTranslation } from 'react-i18next';
+import clsx from 'clsx';
+import { formatDateTime } from 'common/utils/utils';
```
Comment (anton-karlovskiy) [inline src/components/MaintenanceBanner/index.tsx:4]: There can be a line break to separate third-party modules and our own custom modules.
```
import clsx from 'clsx';

import { formatDateTime } from 'common/utils/utils';
```

File: src/components/MaintenanceBanner/index.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+                'md:max-w-3xl',
+                'border',
+                'border-solid',
+                'border-interlayGrey',
+                'rounded',
+                'text-center'
+              )}>
+              <strong className='text-interlayGrey'>
+                {`${maintenance.type} ${formatDateTime(maintenance.startTime)}: ${maintenance.reason} `}
+                <a href={maintenance.link}>{t('maintenance.info')}</a>
```
Comment (anton-karlovskiy) [inline src/components/MaintenanceBanner/index.tsx:49]: Could you please use `InterlayIink`?

File: src/components/MaintenanceBanner/index.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  const { t } = useTranslation();
+  const now = Date.now();
+
+  const maintenanceWindows: Array<Message> = [
+    {
+      startTime: new Date(1616684400000), // Thu Mar 25 2021 15:00:00 GMT+0000
+      endTime: new Date(1616698800000), // Thu Mar 25 2021 19:00:00 GMT+0000
+      type: t('maintenance.type.scheduled'),
+      reason: t('maintenance.reason.chain'),
+      link: 'https://medium.com/interlay/polkabtc-beta-scheduled-maintenance-25-march-3pm-gmt-509b36f73479'
```
Comment (anton-karlovskiy) [inline src/components/MaintenanceBanner/index.tsx:25]: I think we can manage such URLs in `src\config\links.ts`.

File: src/assets/locales/en/translation.json (hunk 1 above)
Comment (nud3l) [reply src/assets/locales/en/translation.json:527]: Hm, the plugin didn't flag it. Good catch, thanks!

File: src/components/MaintenanceBanner/index.tsx (hunk 2 above)
Comment (nud3l) [reply src/components/MaintenanceBanner/index.tsx:11]: funny, the compiler did not complain. You are right though.

File: src/components/MaintenanceBanner/index.tsx (hunk 5 above)
Comment (nud3l) [reply src/components/MaintenanceBanner/index.tsx:25]: In this case, it's only used here and would only be updated here, so I'd keep here for now.

File: src/components/MaintenanceBanner/index.tsx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    }
+  ];
+
+  return (
+    <div>
```
Comment (anton-karlovskiy) [inline src/components/MaintenanceBanner/index.tsx:30]: I think it should be `<ul>` as it's a semantic element and the children should be `<li>`.

File: src/components/MaintenanceBanner/index.tsx [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+      {maintenanceWindows.map(maintenance => {
+        if (now < Number(maintenance.endTime)) {
+          return (
+            <div
```
Comment (anton-karlovskiy) [inline src/components/MaintenanceBanner/index.tsx:36]: I think we do not use `if statement` in favor of ternary statement in JSX.
```
{maintenanceWindows.map(maintenance => (
  <>
    {now < Number(maintenance.endTime) ? (
      <li key='...'>...</li>
    ) : null}
  </>
)}
```

File: src/components/MaintenanceBanner/index.tsx [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+              className={clsx(
+                'px-5',
+                'py-3',
+                'm-4',
+                'sm:mx-auto',
+                'md:max-w-3xl',
```
Comment (anton-karlovskiy) [inline src/components/MaintenanceBanner/index.tsx:42]: I guess `sm:mx-auto` should be `md:mx-auto` as it's full-width smaller than `md`.

File: src/components/MaintenanceBanner/index.tsx [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+              )}>
+              <strong className='text-interlayGrey'>
+                {`${maintenance.type} ${formatDateTime(maintenance.startTime)}: ${maintenance.reason} `}
+                <a href={maintenance.link}>{t('maintenance.info')}</a>
+              </strong>
+            </div>
+          );
+        } else {
+          return (
+            <div></div>
```
Comment (anton-karlovskiy) [inline src/components/MaintenanceBanner/index.tsx:62]: Dangling :)

File: src/parts/Layout/index.tsx [hunk 10]
```diff
@@ -12,6 +12,7 @@ import { PAGES } from 'utils/constants/links';
 import { StoreType } from 'common/types/util.types';
 import styles from './layout.module.css';
 import { ACCOUNT_ID_TYPE_NAME } from '../../constants';
+import MaintenanceBanner from 'components/MaintenanceBanner';
```
Comment (anton-karlovskiy) [inline src/parts/Layout/index.tsx:15]: Grouping imports for components. It's completely minor. :)

File: src/components/MaintenanceBanner/index.tsx (hunk 5 above)
Comment (anton-karlovskiy) [reply src/components/MaintenanceBanner/index.tsx:27]: It sounds reasonable.
I agree that we do not have to move it there in such cases.

File: src/components/MaintenanceBanner/index.tsx [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+}
+
+// TODO: make this easily modifiable by having a list of maintenance elements somewhere else
+function MaintenanceBanner() {
+  const { t } = useTranslation();
+  const now = Date.now();
+
+  const maintenanceWindows: Array<Message> = [
+    {
+      startTime: new Date(1616684400000), // Thu Mar 25 2021 15:00:00 GMT+0000
```
Comment (anton-karlovskiy) [inline src/components/MaintenanceBanner/index.tsx:23]: We can add one pair e.g. `id: 1` and use it as a key.

**Review by anton-karlovskiy — APPROVED** [review]

**Comment by anton-karlovskiy** [comment]
LGTM!

File: src/components/MaintenanceBanner/index.tsx [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+      endTime: new Date(1616698800000), // Thu Mar 25 2021 19:00:00 GMT+0000
+      type: t('maintenance.type.scheduled'),
+      reason: t('maintenance.reason.chain'),
+      link: 'https://medium.com/interlay/polkabtc-beta-scheduled-maintenance-25-march-3pm-gmt-509b36f73479'
+    }
+  ];
+
+  return (
+    <ul>
+      {maintenanceWindows.map(maintenance => (
```
Comment (anton-karlovskiy) [inline src/components/MaintenanceBanner/index.tsx:33]: I would filter first and if the filtered length is zero, then just return null.
Then we can avoid dangling `ul`.

**Pull request merged**